doc/news/changes/minor/20230631Bangerth

```diff
@@ -0,0 +1,6 @@
+Changed: The constructor of the
+PETScWrappers::PreconditionSSOR::AdditionalData class is now
+`explicit`, thereby disallowing the implicit conversion of a number
+(the relaxation factor) to an object of this type.
+<br>
+(Wolfgang Bangerth, 2023/07/01)
```

Comment:
Sorry for nit-picking, but would you mind using the right date (july 1 rather than june 31) in the file name as well?

Reply:
No, because then that will collide with one of my other changelog entries :-)

Can I ask for a variance for this issue given that the CI is already happy? We're so far back with the CI that I don't want to have to re-test this.